src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -83,26 +82,31 @@ class PythonSetup(Subsystem):
 
     @memoized_property
     def interpreter_constraints(self) -> tuple[str, ...]:
-        # TODO: In 2.17.0.dev3 we should set the default above to None and tweak the message here
-        #  appropriately.
-        if self.options.is_default("interpreter_constraints"):
-            warn_or_error(
-                "2.17.0.dev3",
-                "the factory default interpreter constraints value",
+        if not self._interpreter_constraints:
+            # TODO: This is a hacky affordance for Pants's own tests, dozens of which were
+            #  written when Pants provided default ICs, and implicitly rely on that assumption.
+            #  We'll probably want to find and modify all those tests to set an explicit IC, but
+            #  that will take time.
+            if "PYTEST_CURRENT_TEST" in os.environ:
+                return (">=3.7,<4",)
```

Comment:
Maybe you could include another (longer) deprecation here, to commit us to doing that on some timeframe? Maybe 4 releases or something.